main.go

```diff
@@ -64,31 +73,98 @@ func main() {
+	spansIndexedBefore := es.Count(testNode, "apm*span*")
+	transactionsIndexedBefore := es.Count(testNode, "apm*transaction*")
+	errorsIndexedBefore := es.Count(testNode, "apm*error*")
 
-	report, err := w.Work()
+	result, err := w.Work()
 	if err != nil {
 		logger.Errorf(err.Error())
 	}
-	logger.Debugf("%s elapsed since event generation completed", report.Flushed.Sub(report.End))
+	logger.Debugf("%s elapsed since event generation completed", result.Flushed.Sub(result.End))
 
 	fmt.Println()
-	fmt.Println(report)
+	fmt.Println(result)
 
 	metricsAfter, merr2 := server.QueryExpvar(*apmServerSecret, *apmServerUrl)
-	if merr2 == nil {
-		if merr1 == nil {
-			fmt.Println()
-			fmt.Println(metricsAfter.Memstats.Sub(metricsBefore.Memstats))
-		}
-	} else {
-		logger.Errorf("could not get expvar metrics: %s", merr2.Error())
-	}
 
 	info, err := server.QueryInfo(*apmServerSecret, *apmServerUrl)
 	if err != nil {
 		logger.Errorf("apm-server health error: %s", err.Error())
 	} else {
 		fmt.Println(out.Bold + "\n*** " + info.String() + " ***\n" + out.Reset)
 	}
+
+	spansIndexedAfter := es.Count(testNode, "apm*span*")
+	transactionsIndexedAfter := es.Count(testNode, "apm*transaction*")
+	errorsIndexedAfter := es.Count(testNode, "apm*error*")
+
+	this, _ := os.Hostname()
+	r := reports.WithDerivedAttributes(reports.Report{
+		Agent:                          "Go",
+		ReportId:                       shortId(),
+		ReportDate:                     time.Now().Format(reports.GITRFC),
+		ReporterHost:                   this,
+		Timestamp:                      time.Now(),
+		ElasticHost:                    host(*apmElasticsearchUrl),
+		ApmHost:                        host(*apmServerUrl),
+		ApmBuild:                       info.BuildSha,
+		ApmBuildDate:                   info.BuildDate,
+		NumApms:                        1,
+		ApmVersion:                     info.Version,
+		RunTimeout:                     runTimeout.Seconds(),
+		Elapsed:                        result.Flushed.Sub(result.Start).Seconds(),
+		Requests:                       result.NumRequests,
+		FailedRequests:                 result.Errors.SendStream,
+		ErrorsGenerated:                result.ErrorsSent + result.ErrorsDropped,
+		ErrorsSent:                     result.ErrorsSent,
+		ErrorGenerationFrequency:       *errorFrequency,
+		TransactionsGenerated:          result.TransactionsSent + result.TransactionsDropped,
+		TransactionsSent:               result.TransactionsSent,
+		TransactionGenerationFrequency: *transactionFrequency,
+		SpansGenerated:                 result.SpansSent + result.SpansDropped,
+		SpansSent:                      result.SpansSent,
+		EventsAccepted:                 result.Accepted,
+		ErrorsIndexed:                  errorsIndexedAfter - errorsIndexedBefore,
+		TransactionsIndexed:            transactionsIndexedAfter - transactionsIndexedBefore,
+		SpansIndexed:                   spansIndexedAfter - spansIndexedBefore,
```

Comment:
maybe sort these or add space in between sections?  Having a hard time auditing these

Reply:
done in 82807b6b67bbe2c870e147149067b21c91ce51bc, along with the Apm Settings attribute.
also added some cleanup